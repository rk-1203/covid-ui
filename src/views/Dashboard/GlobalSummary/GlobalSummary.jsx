import { getFormattedDate, getFormattedNumber, getFormattedTime } from '../../../utils/utils';
import "./GlobalSummary.scss";

function GlobalSummary({ summary }) {

    const { NewConfirmed, TotalConfirmed, NewDeaths, TotalDeaths, NewRecovered, TotalRecovered, Date } = summary;

    return (
        <section className="global-summary">
            <h1>Global Summary</h1>
            <div className="summary-items">
                <div className="item">
                    <span className="label">New Confirmed Cases</span>
                    {getFormattedNumber(NewConfirmed)}
                </div>
                <div className="item">
                    <span className="label">Total Confirmed Cases</span>
                    {getFormattedNumber(TotalConfirmed)}
                </div>
                <div className="item">
                    <span className="label">New Deaths</span>
                    {getFormattedNumber(NewDeaths)}
                </div>
                <div className="item">
                    <span className="label">Total Deaths</span>
                    {getFormattedNumber(TotalDeaths)}
                </div>
                <div className="item">
                    <span className="label">New Recovered Cases</span>
                    {getFormattedNumber(NewRecovered)}
                </div>
                <div className="item">
                    <span className="label">Total Recovered Cases</span>
                    {getFormattedNumber(TotalRecovered)}
                </div>
                <div className="item">
                    <span className="label">Last Updated Time</span>{`${getFormattedDate(Date)}, ${getFormattedTime(Date)}`}
                </div>
            </div>
        </section>
    );
}

export default GlobalSummary;
