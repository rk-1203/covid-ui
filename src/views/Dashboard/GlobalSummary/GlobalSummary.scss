.global-summary {
    box-shadow: 0 1px 6px 0 rgb(32 33 36 / 28%);
    border-radius: 20px;
    margin-bottom: 40px;
    h1 {
        text-align: left;
        padding: 40px 40px 20px 40px;
        margin: 0;
        font-size: 32px;
    }
    .summary-items {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-row-gap: 20px;
        padding: 20px;
        @media screen and (min-width: 600px) {
            grid-template-columns: 1fr 1fr 1fr;
        }
        .item {
            .label {
                font-weight: bold;
                display: block;
                padding: 10px;
            }
            &:last-child {
                grid-column-start: 1;
                grid-column-end: 3;
                @media screen and (min-width: 600px) {
                    grid-column-start: 1;
                    grid-column-end: 4;
                }
            }
        }
    }
}