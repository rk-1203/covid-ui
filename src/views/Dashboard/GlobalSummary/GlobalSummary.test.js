import React from 'react';
import { render } from '@testing-library/react';

import GlobalSummary from "./GlobalSummary";

const props = {
    summary: {
        NewConfirmed: 1929169,
        TotalConfirmed: 400178665,
        NewDeaths: 9389,
        TotalDeaths: 5760245,
        NewRecovered: 0,
        TotalRecovered: 0,
        Date: "2022-02-09T16:49:33.508Z"
    }
}

describe("GlobalSummary", () => {

    it("renders properly", () => {
        const { container } = render(<GlobalSummary {...props}/>);
        expect(container.firstChild).toMatchSnapshot();
    })

})