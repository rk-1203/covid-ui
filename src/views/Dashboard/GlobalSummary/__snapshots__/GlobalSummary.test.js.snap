// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSummary renders properly 1`] = `
<section
  class="global-summary"
>
  <h1>
    Global Summary
  </h1>
  <div
    class="summary-items"
  >
    <div
      class="item"
    >
      <span
        class="label"
      >
        New Confirmed Cases
      </span>
      19,29,169
    </div>
    <div
      class="item"
    >
      <span
        class="label"
      >
        Total Confirmed Cases
      </span>
      40,01,78,665
    </div>
    <div
      class="item"
    >
      <span
        class="label"
      >
        New Deaths
      </span>
      9,389
    </div>
    <div
      class="item"
    >
      <span
        class="label"
      >
        Total Deaths
      </span>
      57,60,245
    </div>
    <div
      class="item"
    >
      <span
        class="label"
      >
        New Recovered Cases
      </span>
      0
    </div>
    <div
      class="item"
    >
      <span
        class="label"
      >
        Total Recovered Cases
      </span>
      0
    </div>
    <div
      class="item"
    >
      <span
        class="label"
      >
        Last Updated Time
      </span>
      09 Feb 2022, 10:19 PM
    </div>
  </div>
</section>
`;
